### ISSUE TITLE
Model.Array does not work inside Model

### ISSUE THREAD

**Issue by vbarbarosh** [sylvainpolletvillard/ObjectModel#26]
As expected:
```javascript
var Cards = Model.Array([Number, "J","Q","K"]);
var a = new Cards([2]);
a.push(1);
a.push(false); // ERROR
console.log(a);
```

Not as expected:
```javascript
var Container = Model({
    items: Model.Array([Number])
});

var a = new Container({items: []});
a.items.push(1);
a.items.push(false); // OK?
console.log(JSON.stringify(a));
```

**Comment by sylvainpolletvillard** [comment]
Definitely a bug. Thanks for the report

**Comment by sylvainpolletvillard** [comment]
This issue only concerns browsers that support Proxies.

**Comment by sylvainpolletvillard** [comment]
Fixed in v2.1.2. That was a tricky one, related to getPrototypeOf proxy trap. Thanks again for the report
